src/Sign.Core/DataFormatSigners/ClickOnceSigner.cs

```diff
@@ -273,5 +274,53 @@ public void CopySigningDependencies(FileInfo deploymentManifestFile, DirectoryIn
                 }
             }
         }
+
+        /// <summary>
+        /// Try and find the application manifest (.manifest) file from a clickonce application manifest (.application / .vsto
+        /// There might not be one, if the user is attempting to only re-sign the deployment manifest without touching other files.
+        /// This is necessary because there might be multiple *.manifest files present, e.g. if a DLL that's part of the clickonce
+        /// package ships its own assembly manifest which isn't a clickonce application manifest.
+        /// </summary>
+        /// <param name="deploymentManifest"></param>
+        /// <returns>A string containing the file name of the Application manifest, or null if it couldn't be found.</returns>
+        /// <exception cref="InvalidDataException"></exception>
+        private string? GetApplicationManifestForDeploymentManifest(FileInfo deploymentManifest)
+        {
+            var xml = new XmlDocument();
+            xml.Load(deploymentManifest.FullName);
+            // there should only be a single result here, if the file is a valid clickonce manifest.
+            var dependentAssemblies = xml.GetElementsByTagName("dependentAssembly");
+            if (dependentAssemblies.Count != 1)
+            {
+                Logger.LogDebug(Resources.ApplicationManifestNotFound);
+                return null;
+            }
+
+            var node = dependentAssemblies.Item(0);
+            if (node is null || node.Attributes is null)
+            {
+                Logger.LogDebug(Resources.ApplicationManifestNotFound);
+                return null;
+            }
+
+            var applicationManifestFileNameAttribute = node.Attributes["codebase"];
+            if (applicationManifestFileNameAttribute is null || string.IsNullOrEmpty(applicationManifestFileNameAttribute.Value))
+            {
+                Logger.LogDebug(Resources.ApplicationManifestNotFound);
+                return null;
+            }
+
+            // the codebase attribute can be a relative file path (e.g. Application Files\MyApp_1_0_0_0\MyApp.dll.manifest) or
+            // a URI (e.g. https://my.cdn.com/clickonce/MyApp/ApplicationFiles/MyApp_1_0_0_0/MyApp.dll.manifest) so we need to
+            // handle both cases and extract just the file name part.
+            //
+            // we only try and parse absolute uris, because a relative uri can just be treated like a file path for our purposes
+            if (Uri.TryCreate(applicationManifestFileNameAttribute.Value, UriKind.Absolute, out var uri))
+            {
+                Path.GetFileName(uri.LocalPath); // works for http(s) and file:// uris
```

Comment:
There is a bug here.  I think you meant:

```suggestion
                return Path.GetFileName(uri.LocalPath); // works for http(s) and file:// uris
```

Seems worth adding a unit test for this case.  I think you can convert your new, existing unit test into a theory and parameterize `codebase` with both scenarios.